core/src/main/java/org/apache/calcite/rel/rules/AggregateReduceFunctionsRule.java

```diff
@@ -94,15 +94,23 @@
 
   /** The singleton. */
   public static final AggregateReduceFunctionsRule INSTANCE =
-      new AggregateReduceFunctionsRule(operand(LogicalAggregate.class, any()),
+      new AggregateReduceFunctionsRule(operand(LogicalAggregate.class, any()), true,
           RelFactories.LOGICAL_BUILDER);
 
+  /** The singleton. */
+  public static final AggregateReduceFunctionsRule NO_REDUCE_SUM =
+          new AggregateReduceFunctionsRule(operand(LogicalAggregate.class, any()), false,
+                                           RelFactories.LOGICAL_BUILDER);
+
+  private final boolean reduceSum;
+
   //~ Constructors -----------------------------------------------------------
 
   /** Creates an AggregateReduceFunctionsRule. */
-  public AggregateReduceFunctionsRule(RelOptRuleOperand operand,
+  public AggregateReduceFunctionsRule(RelOptRuleOperand operand, boolean reduceSum,
```

Comment:
Changing a public constructor brakes backward compatibility. I don't think this is desired for this reason.

Reply:
Moreover, I don't like a lot the idea of making the rule configurable **just** for a single function (i.e., SUM). If we go this direction, I would prefer to be able to configure the behavior of this rule for all functions that it handles.

Reply:
To address the backward compatibility part, is it okay to add this as a new constructor as opposed to changing the current constructor?

Reply:
It is ok to add a new constructor if it is well-documented and general enough to handle more than just the SUM function.

Reply:
done.

Reply:
Thanks, I added a few more comments.